core/src/WsTrust/MexParser.cs

```diff
@@ -0,0 +1,272 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Identity.Core.Http;
+
+namespace Microsoft.Identity.Core.WsTrust
```

Comment:
This is a bit difficult to review because it is not clear if you added / changed any code here - I see you added some comments. Most of the code already exists so it should not be reviewed. Not sure how to overcome this issue as we will have many similar PRs in the future ...

Reply:
Today I went through all the previous WIA codes and test cases in ADAL, confirmed that they are all covered by the new implementation. So I delete them now.
Git(hub) seems smart enough to display the addition AND deletion AS a modification. This makes further review much easier. Yay!

Reply:
Glad that Bogdan also has the mindset of "the code already exists ... should not be reviewed". :-)  We should definitely discuss this philosophy in next retrospect.

Meanwhile, quoted from my earlier reply from Github:

Today I went through all the previous WIA codes and test cases in ADAL, confirmed that they are all covered by the new implementation. So I delete them now.
Git(hub) seems smart enough to display the addition AND deletion AS a modification. This makes further review much easier. Yay!

---
In reply to: [203666521](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r203666521) [](ancestors = 203666521)